compiler/src/parsing/parsetree.re

```diff
@@ -458,7 +459,7 @@ and expression_desc =
   | PExpConstraint(expression, parsed_type)
   | PExpLambda(list(pattern), expression)
   | PExpApp(expression, list(expression))
-  | PExpConstruct(loc(Identifier.t), list(expression))
+  | PExpConstruct(loc(Identifier.t), list(expression), bool) // bool indicates whether this is an inline record constructor or not
```

Comment:
Hmm, I actually thought about doing this initially but went with this way because it seemed that as a whole it would make this type less cumbersome to use. I suppose I do agree that doing it the ADT way over the boolean flag way would make it more clear where this distinction really matters though.